imagegw/test/imagemngr_test.py

```diff
@@ -892,6 +893,48 @@ def test_pull_acl(self):
         self.images.drop()
         self.stop_worker()
 
+    def test_import(self):
+        """
+        Basic import test including an induced pull failure.
+        """
+        # Use defaults for format, arch, os, ostcount, replication
+        pr = {
+            'system': self.system,
+            'itype': self.itype,
+            'tag': self.tag,
+            'remotetype': 'dockerv2',
+            'filepath': '/images/test/test.squashfs',
+            'format': 'squashfs',
+            'userACL': [],
+            'groupAcl': []
+        }
+        # Do the pull
+        self.start_worker()
+        session = self.m.new_session(self.auth, self.system)
+        rec = self.m.mngrimport(session, pr, testmode=1)  # ,delay=False)
+        assert rec is not None
+        id = rec['_id']
+        # Confirm record
+        q = {'system': self.system, 'itype': self.itype, 'pulltag': self.tag}
+        mrec = self.images.find_one(q)
+        assert '_id' in mrec
+        # Track through transistions
+        state = self.time_wait(id)
+        #Debug
+        assert state == 'READY'
+        imagerec = self.m.lookup(session, pr)
+        assert 'ENTRY' in imagerec
+        assert 'ENV' in imagerec
+        # Cause a failure
+        self.images.drop()
+        rec = self.m.pull(session, pr, testmode=2)
```

Comment:
We could probably cut below this except for the stop worker.

Reply:
I didn't change this part for my code, but I can cut it out if you want.